Source/WebKit/WebProcess/WebPage/WebFoundTextRangeController.cpp

```diff
@@ -151,10 +151,10 @@ void WebFoundTextRangeController::decorateTextRangeWithStyle(const WebFoundTextR
     if (auto simpleRange = simpleRangeFromFoundTextRange(range)) {
         switch (style) {
         case FindDecorationStyle::Normal:
-            simpleRange->start.document().markers().removeMarkers(*simpleRange, WebCore::DocumentMarkerType::TextMatch);
+            simpleRange->start.protectedDocument()->checkedMarkers()->removeMarkers(*simpleRange, WebCore::DocumentMarkerType::TextMatch);
```

Comment:
This is interesting. This seems to be the warning related to this line: https://build.webkit.org/results/Apple-Sequoia-Safer-CPP-Checks/292329@main%20(2568)/scan-build-output/StaticAnalyzer/WebKit/StaticAnalyzerReports/report-WebFoundTextRangeController.cpp-decorateTextRangeWithStyle-15-ebd5f3.html#EndPath

1. The analyzer says `document().markers()` is unsafe. But `markers()` is inline.
2. It doesn't flag `markers().removeMarkers()` as unsafe even though `removeMarkers()` is not trivial. 

Aditya's fix here seems perfect but do we want to file bugs for the analyzer here @rniwa?

Reply:
@RupinMittal While `markers()` is inline, it is not trivial because it calls a non-inline function: `ensureMarkers()`